src/main/java/aeneas/models/Level.java

```diff
@@ -30,6 +31,9 @@ public void setRefreshListener(RefreshListener listener) {
   transient int levelNumber;
   transient boolean active = false;
   boolean prebuilt;
+  
+  transient Stack<IMove> undoStack;
```

Comment:
If a level is opened from a serialized file, `undoStack` and `redoStack` will be null. This means that as soon as anyone attempts to make a move, a NullPointerException is generated.